### PR TITLE
FEAT(day_10): Add go_forward implementation

### PR DESCRIPTION
**Pull request by dariannax3**

### CODE REVIEW

File: day_10/task_1/pipe_maze.cpp [hunk 1]
```diff
@@ -104,3 +104,19 @@ std::pair<MazePoint, MazePoint> find_next_neighbors(const PipeMaze& pipe_maze,
...
+    {
+        return foundNeighbors.first;
+    }
+    if(foundNeighbors.second != previous_point)
+    {
+        return foundNeighbors.second;
+    }
+    
+    return MazePoint(-1, -1);
+}
```
Comment (Rosein) [inline day_10/task_1/pipe_maze.cpp]: Missing newline

File: day_10/task_1/pipe_maze.hpp [hunk 2]
```diff
@@ -8,3 +8,6 @@ using MazePoint = std::pair<int, int>;
 MazePoint find_starting_point(const PipeMaze& pipe_maze);
 std::pair<MazePoint, MazePoint> find_next_neighbors(const PipeMaze& pipe_maze,
                                                     const MazePoint& maze_point);
+MazePoint go_forward(const PipeMaze& pipe_maze, 
+                     const MazePoint& current_point, 
+                     const MazePoint& previous_point);
```
Comment (Rosein) [inline day_10/task_1/pipe_maze.hpp]: Add missing newline

File: day_10/task_1/pipe_maze.cpp [hunk 3]
```diff
@@ -104,3 +104,19 @@ std::pair<MazePoint, MazePoint> find_next_neighbors(const PipeMaze& pipe_maze,
...
+    if(foundNeighbors.first != previous_point)
+    {
+        return foundNeighbors.first;
+    }
+    if(foundNeighbors.second != previous_point)
+    {
+        return foundNeighbors.second;
+    }
+    
+    return MazePoint(-1, -1);
```
Comment (Rosein) [inline day_10/task_1/pipe_maze.cpp]: Name magic object as kSomeErrorInformation, but please use more specific description to express the branch flow case.

**Review by Rosein — CHANGES_REQUESTED** [review]
Almost good. I have only few minor suggestions :)

File: day_10/task_1/pipe_maze.cpp (hunk 1 above)
Comment (dariannax3) [reply day_10/task_1/pipe_maze.cpp]: done

File: day_10/task_1/pipe_maze.hpp (hunk 2 above)
Comment (dariannax3) [reply day_10/task_1/pipe_maze.hpp]: done

File: day_10/task_1/pipe_maze.cpp (hunk 3 above)
Comment (dariannax3) [reply day_10/task_1/pipe_maze.cpp]: done

File: day_10/task_1/pipe_maze.cpp [hunk 4]
```diff
@@ -2,6 +2,8 @@
 #include <cassert>
 #include <string>
 
+MazePoint invalidMazeEntryPoint(-1, -1);
```
Comment (Rosein) [inline day_10/task_1/pipe_maze.cpp]: Please add constexpr or const to type and change brackets into curly brackets :)

**Review by Rosein — CHANGES_REQUESTED** [review]
Last one minor suggestion :)

File: day_10/task_1/pipe_maze.cpp (hunk 4 above)
Comment (dariannax3) [reply day_10/task_1/pipe_maze.cpp]: done

**Review by Rosein — APPROVED** [review]
Good job!

**Pull request merged**